src/net.cpp

```diff
@@ -2801,6 +2801,9 @@ std::vector<CAddress> CConnman::GetAddresses(CNode& requestor, size_t max_addres
     uint64_t cache_id = GetDeterministicRandomizer(RANDOMIZER_ID_ADDRCACHE)
         .Write(requestor.ConnectedThroughNetwork())
         .Write(local_socket_bytes.data(), local_socket_bytes.size())
+        // For outbound connections, the port of the bound address is randomly
+        // assigend by the OS and would therefore not be useful for seeding.
+        .Write(requestor.IsInboundConn() ? requestor.addrBind.GetPort() : 0)
```

Comment:
Before this change, all inbound onion connections get the same cache_id. After this change, every different onion would generate a new cache_id. Am i right?

Reply:
> Before this change, all inbound onion connections get the same cache_id.

Yes and all other incoming connections to 127.0.0.1:* also get the same cache_id.

> After this change, every different onion would generate a new cache_id.

After this change all incoming connections to 127.0.0.1 binds get the same cache_id if the bound port is the same.
e.g.: All connections to 127.0.0.1:8333 will get the same cache_id "a" and all connections to 127.0.0.1:8334 will get the same cache_id "b".

Reply:
I assume the bound port would be the same for all Tor connections, right?

Reply:
You can have more than one onion bind (not the default) by using `-bind=<addr>=onion` multiple times. In that case not all Tor connections would have the same bound port.

example:
```
-bind=127.0.0.1:5555=onion
-bind=127.0.0.1:6666=onion
```
With this config any inbound Tor connection will have `addrBind` set to `127.0.0.1:5555` or `127.0.0.1:6666`.